src/caselawclient/models/judgments.py

```diff
@@ -23,66 +23,101 @@
 JUDGMENT_STATUS_IN_PROGRESS = "In progress"
 
 
+class MarkLogicAttributeDict(TypedDict):
+    getter: Callable[[str], Union[str, bool]]
+    setter: None
+
+
 class CannotPublishUnpublishableJudgment(Exception):
     pass
 
 
 class Judgment:
+    MARKLOGIC_ATTRIBUTE_MAP: dict[str, MarkLogicAttributeDict]
+
+    values_store: dict[str, Union[str, bool]] = {}
+
+    # These type definitions make sure that downstream users of this class can rely on the types
+    uri: str
+    api_client: MarklogicApiClient
+    name: str
+    neutral_citation: str
+    court: str
+    judgment_date_as_string: str
+    is_published: bool
+    is_sensitive: bool
+    is_anonymised: bool
+
     def __init__(self, uri: str, api_client: MarklogicApiClient):
         self.uri = uri.strip("/")
         self.api_client = api_client
         if not self.judgment_exists():
             raise JudgmentNotFoundError(f"Judgment {self.uri} does not exist")
 
+        # Callables for setter accept a single argument of the judgment URI
+        self.MARKLOGIC_ATTRIBUTE_MAP = {
+            "name": {
+                "getter": self.api_client.get_judgment_name,
+                "setter": None,
+            },
+            "neutral_citation": {
```

Comment:
I don't think I'm quite grasping what's going on with this attr override.

Quick call after standup to go through it?